### DISCUSSION TITLE
Why "vcpkg install" in manifest mode removes previous installed triplet ?

### DISCUSSION THREAD

**Discussion by freedbrt** [microsoft/vcpkg#32631] [General]
I have a simple vcpkg.json file
```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
  "name": "app",
  "version-string": "1.0.0",
  "dependencies": [ ... ]
}
```

When i trying to run "vcpkg install --triplet A" everything is fine, all my packges will be installed in vcpkg_installed folder. But when i run "vcpkg install --triplet B" second time, it removed my previous installed packages from triplet A, why it happenes?
In classic mode it won't happened.

When i run "vcpkg install --triplet B" second time, I see
```
root@root vcpkg % ./vcpkg install --triplet B
The following packages will be removed:
    // here list of packages of triplet A
The following packages will be built and installed:
    // here list of packages of triplet B
```

**Comment by autoantwort** [comment]
Because the semantic of `install` in manifest mode is _Set the installed dir to the list of ports in the vcpkg.json file for the specified triplet_. In the background it is implemented via `x-set-installed`.

**Reply by freedbrt** [reply]
@autoantwort Is it possible to change this behavior? Because the only way now to do that - install different triplets into different folders via -x-install-root, and than merge it into one using script or something like that.

**Reply by autoantwort** [reply]
No this is not possible. Why do you want multiple triplets in the same installed folder?

**Reply by autoantwort** [reply]
You could use `vcpkg x-set-installed sqlite:x64-windows sqlite:x64-windows-static sqlite:x86-windows`

**Reply by freedbrt** [reply]
Because i trying to build my dependencies using 4 triplets arm64-android, arm-android, x64-android, x86-android.
And before I used a classic way to install libraries "vcpkg install libA libB --triplet ..." and everything was in installed folder.
Now I wrote vcpkg.json with all my dependencies, and trying to do the same, but for me it was a surprise that each new "vcpkg install" call will delete the previous triplet. I must now do the trick so that the installation is done in different folders using -x-install-root, but its not convenient for me, and I need to rewrite my build script.

**Reply by Neumann-A** [reply]
> Because i trying to build my dependencies using 4 triplets arm64-android, arm-android, x64-android, x86-android.

This would imply for different build dirs, wouldn't it? Manifest mode is intended to install into the build dir which avoids the problem you describe.

**Reply by freedbrt** [reply]
I put my vcpkg and vcpkg_install into source folder, not build folder. I understand that the manifest mode is not designed for my use case, I'll switch back to the classic mode.
Thanks for the answers.

**Reply by Jochen0x90h** [reply]
I also wonder why it is not possible to install multiple triplets into the same directory. The CMake toolchain should be able to select the right libs based on the current triplet. Now I do

vcpkg.exe install --triplet x64-windows --x-install-root x64-windows

and then set "VCPKG_INSTALLED_DIR": "x64-windows" in my CMakeUserPresets.json

**Reply by ovidiubuligan** [reply]
I really want to use the `vcpkg.json` since it makes life a lot  simpler . Even with let's say 2 custom install directories for 2 different triplets like `vcpkg install --x-install-root="otel_x64_install" --triplet x64-windows `  and `vcpkg install --x-install-root="otel_x86_install" --triplet x86-windows`   , I can't use the `vcpkg export --nuget`  .
 Is there a way to export a nuget that containes x86 and x64 triplets in `vcpkg.json` mode ?
